pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -19,40 +19,58 @@ package sidecarmetricscollector
 import (
 	"os"
 	"path/filepath"
-	"reflect"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	commonv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 )
 
-var testJsonDataPath = filepath.Join("testdata", "JSON")
+const (
+	validJSONTestFile            = "good.json"
+	invalidFormatJSONTestFile    = "invalid-format.json"
+	invalidTimestampJSONTestFile = "invalid-timestamp.json"
+	missingMetricJSONTestFile    = "missing-objective-metric.json"
 
-func TestCollectObservationLog(t *testing.T) {
+	validTEXTTestFile            = "good.log"
+	invalidValueTEXTTestFile     = "invalid-value.log"
+	invalidFormatTEXTTestFile    = "invalid-format.log"
+	invalidTimestampTEXTTestFile = "invalid-timestamp.log"
+	missingMetricTEXTTestFile    = "missing-objective-metric.log"
+)
 
-	if err := generateTestFiles(); err != nil {
+var (
+	testDir          = "testdata"
+	testJsonDataPath = filepath.Join(testDir, "JSON")
+	testTextDataPath = filepath.Join("testdata", "TEXT")
```

Comment:
You might want to re-use your const here.
```suggestion
	testTextDataPath = filepath.Join(testDir, "TEXT")
```

Reply:
Sorry, I missed this.